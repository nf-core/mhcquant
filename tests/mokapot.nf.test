nextflow_pipeline {

    name "Test pipeline"
    script "../main.nf"
    tag "pipeline"
    tag "test_mokapot"
    profile "test_mokapot"

    test("-profile test_mokapot") {

        when {
            params {
                outdir = "$outputDir"
            }
        }

        then {
            // stable_name: All files + folders in ${params.outdir}/ with a stable name
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            // stable_path: All files in ${params.outdir}/ with stable content
			// Mokapot seems to return non-deterministic results eventhough the random number generator (rng) is fixed
            def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore', ignore: ['*.tsv'])
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    // Number of successful tasks
                    workflow.trace.succeeded().size(),
                    // pipeline versions.yml file for multiqc from which Nextflow version is removed because we test pipelines on multiple Nextflow versions
                    removeNextflowVersion("$outputDir/pipeline_info/nf_core_mhcquant_software_mqc_versions.yml"),
                    // All stable path name, with a relative path
                    stable_name,
                    // All files with stable contents
                    stable_path
                ).match() }
            )
        }
    }
}
