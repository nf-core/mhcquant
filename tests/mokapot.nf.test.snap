{
    "-profile test_mokapot": {
        "content": [
            20,
            {
                "MS2RESCORE": {
                    "MS\u00b2Rescore": "3.1.5)"
                },
                "OPENMSTHIRDPARTY_COMETADAPTER": {
                    "CometAdapter": "3.4.0",
                    "Comet": "2024.01 rev. 1"
                },
                "OPENMS_DECOYDATABASE": {
                    "openms": "3.4.0"
                },
                "OPENMS_IDFILTER_Q_VALUE": {
                    "openms": "3.4.0"
                },
                "OPENMS_IDMASSACCURACY": {
                    "OpenMS": "3.4.0"
                },
                "OPENMS_IDMERGER": {
                    "openms": "3.4.0"
                },
                "OPENMS_IDSCORESWITCHER": {
                    "openms": "3.4.0"
                },
                "OPENMS_PEPTIDEINDEXER": {
                    "openms": "3.4.0"
                },
                "OPENMS_TEXTEXPORTER": {
                    "openms": "3.4.0"
                },
                "PYOPENMS_CHROMATOGRAMEXTRACTOR": {
                    "python": "3.13.1",
                    "pyopenms": "3.3.0"
                },
                "SUMMARIZE_RESULTS": {
                    "python": "3.13.1",
                    "pyopenms": "3.3.0"
                },
                "Workflow": {
                    "nf-core/mhcquant": "v3.1.0dev"
                }
            },
            [
                "HepG2_A.tsv",
                "intermediate_results",
                "intermediate_results/comet",
                "intermediate_results/comet/HepG2_rep1_small_pin.tsv",
                "intermediate_results/comet/HepG2_rep2_small_pin.tsv",
                "intermediate_results/comet/HepG2_rep3_small_pin.tsv",
                "intermediate_results/rescoring",
                "intermediate_results/rescoring/HepG2_A_ms2rescore.idXML",
                "intermediate_results/rescoring/HepG2_A_pout_filtered.idXML",
                "multiqc",
                "multiqc/ms2rescore",
                "multiqc/multiqc_data",
                "multiqc/multiqc_data/BETA-multiqc.parquet",
                "multiqc/multiqc_data/multiqc.log",
                "multiqc/multiqc_data/multiqc_chromatogram.txt",
                "multiqc/multiqc_data/multiqc_citations.txt",
                "multiqc/multiqc_data/multiqc_data.json",
                "multiqc/multiqc_data/multiqc_general_stats.txt",
                "multiqc/multiqc_data/multiqc_histogram_mz.txt",
                "multiqc/multiqc_data/multiqc_histogram_rt.txt",
                "multiqc/multiqc_data/multiqc_histogram_scores.txt",
                "multiqc/multiqc_data/multiqc_length_dist.txt",
                "multiqc/multiqc_data/multiqc_mass_error.txt",
                "multiqc/multiqc_data/multiqc_scores_xcorr.txt",
                "multiqc/multiqc_data/multiqc_software_versions.txt",
                "multiqc/multiqc_data/multiqc_sources.txt",
                "multiqc/multiqc_plots",
                "multiqc/multiqc_plots/pdf",
                "multiqc/multiqc_plots/pdf/chromatogram-cnt.pdf",
                "multiqc/multiqc_plots/pdf/chromatogram-log.pdf",
                "multiqc/multiqc_plots/pdf/general_stats.pdf",
                "multiqc/multiqc_plots/pdf/histogram_mz.pdf",
                "multiqc/multiqc_plots/pdf/histogram_rt.pdf",
                "multiqc/multiqc_plots/pdf/histogram_scores.pdf",
                "multiqc/multiqc_plots/pdf/length_dist.pdf",
                "multiqc/multiqc_plots/pdf/mass_error.pdf",
                "multiqc/multiqc_plots/pdf/scores_xcorr.pdf",
                "multiqc/multiqc_plots/png",
                "multiqc/multiqc_plots/png/chromatogram-cnt.png",
                "multiqc/multiqc_plots/png/chromatogram-log.png",
                "multiqc/multiqc_plots/png/general_stats.png",
                "multiqc/multiqc_plots/png/histogram_mz.png",
                "multiqc/multiqc_plots/png/histogram_rt.png",
                "multiqc/multiqc_plots/png/histogram_scores.png",
                "multiqc/multiqc_plots/png/length_dist.png",
                "multiqc/multiqc_plots/png/mass_error.png",
                "multiqc/multiqc_plots/png/scores_xcorr.png",
                "multiqc/multiqc_plots/svg",
                "multiqc/multiqc_plots/svg/chromatogram-cnt.svg",
                "multiqc/multiqc_plots/svg/chromatogram-log.svg",
                "multiqc/multiqc_plots/svg/general_stats.svg",
                "multiqc/multiqc_plots/svg/histogram_mz.svg",
                "multiqc/multiqc_plots/svg/histogram_rt.svg",
                "multiqc/multiqc_plots/svg/histogram_scores.svg",
                "multiqc/multiqc_plots/svg/length_dist.svg",
                "multiqc/multiqc_plots/svg/mass_error.svg",
                "multiqc/multiqc_plots/svg/scores_xcorr.svg",
                "multiqc/multiqc_report.html",
                "pipeline_info",
                "pipeline_info/nf_core_mhcquant_software_mqc_versions.yml"
            ],
            [
                
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-06-20T11:56:49.454075518"
    }
}